packages/react/src/primitives/StepperField/useStepper.ts

```diff
@@ -62,25 +88,37 @@ export const useStepper = ({
 
   // This is the exact value to be rendered on screen
   // It could be a string, like '-' or empty string when users clear the input
-  const [inputValue, setInputValue] = React.useState<number | string>(value);
+  const [inputValue, setInputValue] = React.useState<InputValue>(value);
 
   const handleOnChange: React.ChangeEventHandler<HTMLInputElement> =
-    React.useCallback((event) => {
-      setInputValue(event.target.value);
-    }, []);
+    React.useCallback(
+      (event) => {
+        setInputValue(event.target.value);
+
+        if (typeof onChange === 'function') {
+          onChange(event);
```

Comment:
Just a note that this is a bug fix, previous to this PR `onChange` props passed to `StepperField` was never called

Reply:
Good catch